addons/pos_loyalty/static/src/js/Loyalty.js

```diff
@@ -1632,12 +1631,20 @@ patch(Order.prototype, "pos_loyalty.Order", {
                 this._updateRewards();
             }
         }
+        if (!rule) {
+            return _.str.sprintf(
+                _t("Gift Card %s, balance: %s%s"),
+                code,
+                this.pos.currency.symbol,
+                this.pos.get_order().codeActivatedCoupons.at(-1).balance
```

Comment:
I think the input `coupon` is not always the same as the last item of the `codeActivatedCoupons`. It would be more correct to find the correct coupon from the items then get the balance.